src/table/virtualized-table/Body.tsx

```diff
@@ -52,44 +67,87 @@ const Body = (props: BodyProps) => {
     pageInfo,
   });
 
-  const itemData = useMemo(
-    () => ({ rowsInPage, columns, bodyStyle, isHoverEnabled }),
-    [rowsInPage, columns, bodyStyle, isHoverEnabled]
+  const itemData = useMemo<ItemData>(
+    () => ({ rowsInPage, columns, bodyStyle, isHoverEnabled, maxLineCount }),
+    [rowsInPage, columns, bodyStyle, isHoverEnabled, maxLineCount]
   );
 
+  useEffect(() => {
+    syncHeight(innerForwardRef.current?.clientHeight ?? 0, true);
+  }, [deferredRowCount, syncHeight, innerForwardRef]);
+
+  useEffect(() => {
+    syncHeight(innerForwardRef.current?.clientHeight ?? 0, false);
+
+    if (rowMeta.current.lastScrollToRatio === 1) {
+      // Hack to deal with the case when a user scrolls to the last row
+      // and data has not yet finished loading
+      gridRef.current?.scrollToItem({ rowIndex: deferredRowCount - 1, align: 'start' });
+    }
+  });
+
   useSelectionsEffect(rowsInPage);
 
   useLayoutEffect(() => {
-    if (!forwardRef.current) return;
+    if (!gridRef.current) return;
+
+    rowMeta.current.lastScrollToRatio = 0;
+    gridRef.current.resetAfterIndices({ columnIndex: 0, rowIndex: 0, shouldForceUpdate: true });
+    gridRef.current.scrollTo({ scrollLeft: 0, scrollTop: 0 });
+  }, [layout, pageInfo.page, gridRef, columnWidth, rowMeta, theme.name()]); // eslint-disable-line react-hooks/exhaustive-deps
 
-    forwardRef.current.resetAfterIndices({ columnIndex: 0, rowIndex: 0, shouldForceUpdate: true });
-    forwardRef.current.scrollTo({ scrollLeft: 0, scrollTop: 0 });
-  }, [layout, pageInfo.page, forwardRef, columnWidth]);
+  const bodyHeight = getBodyHeight(
+    rect,
+    headerAndTotalsHeight,
+    deferredRowCount,
+    estimatedRowHeight,
+    rowMeta.current.totalHeight
+  );
 
-  let { height: bodyHeight } = rect;
-  bodyHeight -= headerAndTotalsHeight;
-  bodyHeight = Math.min(deferredRowCount * rowHeight, bodyHeight);
+  useImperativeHandle(
+    ref,
+    () => {
+      return {
+        interpolatedScrollTo: (scrollTopRatio: number, scrollLeft: number) => {
```

Comment:
so this is where you actually scroll based on the interpolated value?

Reply:
or I mean, it is when this runs it scrolls, but this is called from use-scroll-handler

Reply:
Yes, sound like you have figured it out 😃 

But the reason that I use `useImperativeHandle` here instead of just doing this logic directly in the scroll handler, is because of the dependencies. Some of them would have to be passed up to the parent of `Body` to be able to run this logic there. So I figured it was easier to just do like this.